test/test_util.py

```diff
@@ -95,6 +96,8 @@ def test_invalid_host(self):
             'http://google.com:foo',
             'http://::1/',
             'http://::1:80/',
+            'http://google.com:-80',
+            six.u('http://google.com:\xb2\xb2'),  # \xb2 = ^2
```

Comment:
It seems like we don't have tests for this change in Retry-After: can we check that?